lms/djangoapps/learner_dashboard/programs.py

```diff
@@ -242,3 +159,83 @@ def standalone_page_title(self, request, fragment, **kwargs):
         Return page title for the standalone page.
         """
         return _('Program Details')
+
+
+class ProgramDiscussionLTI:
+    """
+      Render the program discussion iframe.
+    """
+    DEFAULT_ROLE = 'Student'
+    ADMIN_ROLE = 'Administrator'
+
+    def __init__(self, program_uuid, request):
+        self.program_uuid = program_uuid
+        self.request = request
+        self.enabled = program_discussions_is_enabled()
+
+    def get_program_discussion_configuration(self):
+        return ProgramDiscussionsConfiguration.objects.filter(
+            program_uuid=self.program_uuid
+        ).first()
+
+    def _get_resource_link_id(self) -> str:
+        site = get_current_site(self.request)
+        return f'{site.domain}-{self.program_uuid}'
+
+    def _get_result_sourcedid(self, resource_link_id) -> str:
+        return f'{self.program_uuid}:{resource_link_id}:{self.request.user.id}'
+
+    def get_user_role(self):
+        """
+        Returns the given user's role
+        """
+        if GlobalStaff().has_user(self.request.user):
+            return self.ADMIN_ROLE
+        return self.DEFAULT_ROLE
```

Comment:
Here is what we can do so that the code is easier to extend in the future. 

````
def get_user_roles(self):
    """
    Returns comma-separated roles for the given user
    """
    basic_role = self.DEFAULT_ROLE

    if GlobalStaff().has_user(self.request.user):
        basic_role = self.ADMIN_ROLE

    all_roles = [basic_role]
    return ','.join(all_roles)
````